app/ModelFunctions/AlbumFunctions.php

```diff
@@ -41,15 +41,26 @@ public function is_smart_album($albumID)
 	 */
 	public function create(string $title, int $parent_id): Album
 	{
-		$num = Album::where('id', '=', $parent_id)->count();
-		// id cannot be 0, so by definition if $parent_id is 0 then...
```

Comment:
True but it is safer: What if you create an album with a wrong `$parent_id`?

Reply:
I'm amused that the line you picked is your own comment, but I see your point. Though this may go back to the other question of how much checking do we want to do for bad data... I guess if one user/browser/session deleted an album between loading the page and creating the subalbum this could be required..

Reply:
I see that you rewrote this code in the meantime, but here was my thinking last night:

1. If `$parent_id` is wrong then we'll never get here because the `UploadCheck` middleware will flag it. Although now that I think about it, it will let it right through for the admin.

2. The test seemed inconsistent with the rest of the code. It seems to me that in most other places if an album doesn't exist, we fail hard. Here we had this weird fallback where we instead create the album at the top level. This is a case that's not supposed to happen so why not fail and return an error to the user?

Reply:
I'm fine with failing hard too. :)